test/run_test.py

```diff
@@ -1356,12 +1361,9 @@ def success_callback(err_message):
             if can_run_in_pytest(test):
                 options_clone.pytest = True
             err_message = run_test_module(test, test_directory, options_clone)
-            if err_message is None:
-                continue
-            failure_messages.append(err_message)
-            if not options_clone.continue_through_error:
+            test_failed = handle_error_messages(err_message)
+            if test_failed and not options.continue_through_error:
```

Comment:
Do you also need to have `failure_messages.append(err_message)` here in the if condition before raising the RuntimeError?  Probably not as it looks like it's not used

Reply:
That bit happens inside `handle_error_messages`